client/lib/use-in-view/index.ts

```diff
@@ -39,7 +39,15 @@ export function useInView< T extends Element >( oneTimeCallback: () => void ): R
 		observerRef.current.observe( elementRef.current );
 
 		// When the effect is dismounted, stop observing
-		return () => observerRef.current?.disconnect?.();
+		return () => {
+			observerRef.current?.disconnect?.();
+			if ( ! viewedRef.current && observerRef.current ) {
+				// This case prevents when the effect is dismounted and re-mounted due oneTimeCallback is not wrapped in useCallback
+				// We cannot show a warning, because it can be a valid use case.
+				// Valid case: When a component is unmounted and never was visible.
+				observerRef.current = undefined;
+			}
+		};
```

Comment:
Should this have tests?

Reply:
Sure, I'll write those tests 👌 .

Reply:
Just so it's stated: I still think we should [use the well-tested dependency](https://github.com/Automattic/wp-calypso/pull/69119#pullrequestreview-1143893128) so we don't have to discover all of these edge cases ourselves.

Reply:
I like our implementation. The `IntersectionObserver` is not something that is evolving constantly, so we don't need to worry about maintenance or finding more edge cases.

Reply:
@danielbachhuber, test added here https://github.com/Automattic/wp-calypso/pull/69378/commits/3f13254be8c58146174569be9d21c90f3033f276